### DISCUSSION TITLE
Pop3Server based on SmtpServer Code

### DISCUSSION THREAD

**Discussion by Lakerfield** [cosullivan/SmtpServer#224] [Show and tell]
Hello @cosullivan,

First of all, I want to express my appreciation for your work on the SmtpServer project. It's a well-crafted piece of software that has been incredibly useful for many.

I wanted to let you know that, as a proof of concept, I have developed a POP3 protocol server by adapting your SmtpServer code. The project is available on [Lakerfield/Pop3Server](https://github.com/Lakerfield/Pop3Server). My implementation follows the same principles and structure as your SMTP server, but, of course, with the necessary modifications to support the POP3 protocol.

Given that this project essentially started as a direct copy of your codebase, I wanted to ask if you would be interested in either:

- Integrating the POP3 implementation into your SmtpServer project, possibly under a unified library that supports both SMTP and POP3 protocols.
- Keeping the projects separate but perhaps establishing a formal linkage or collaboration between them, so that both can benefit from shared updates and improvements.

I am open to discussing how we might collaborate further and would love to get in touch to explore these ideas. Please let me know your thoughts!

Looking forward to your feedback.

Best regards,
Michael Lakerveld

**Comment by Lakerfield** [comment]
I wanted to follow up on my previous message with some additional details about the POP3 server implementation I mentioned.

As this project is an adaptation of your SmtpServer code, you'll notice that many objects and classes still retain the SmtpXxxx prefix. Ideally, these should be renamed to Pop3Xxxx to better align with the POP3 protocol. However, I’ve intentionally left these names unchanged for now to keep the codebase more consistent with your original project.

By maintaining the original naming, I believe it might actually make a potential merge or code-sharing effort easier, as it highlights the shared structure and logic between the two implementations. This should simplify the process of identifying and integrating the similarities and differences between the protocols.

**Comment by WaldenL** [comment]
Not that I get a vote in this, but I’d argue against merging them.

It’s nice to have a component that is only SMTP and not have to worry about
a potential attack vector through POP3. Yes, I know, if we don’t use the
POP3 stuff it’s not an attack vector, but let me introduce you to our
auditors… :)

-W

Sent on the go, please excuse typos and terseness.

**Comment by tinohager** [comment]
I don't see any advantages in merging them either. It would be better to consider which objects form the core and perhaps move them to a new project. The decision is of course up to the maintainer.
